src/load_inputs/load_dataframe.jl

```diff
@@ -127,9 +127,13 @@ function check_for_duplicate_keys(path::AbstractString)
     end
 end
 
-function load_dataframe_from_file(path)::DataFrame
+function load_dataframe_from_file(path::AbstractString)
     check_for_duplicate_keys(path)
```

Comment:
@mmutic and @cfe316 the `check_for_duplicate_keys` function assumes a csv file. Maybe use DuckDB's [`DESCRIBE`](https://duckdb.org/docs/guides/meta/describe.html) function to get column names.

Here's a potential usage. The `con` object could also be passed in from `load_dataframe_from_file` -- not sure if it makes much difference.
```julia
function check_for_duplicate_keys(path::AbstractString)
    con = DBInterface.connect(DuckDB.DB, ":memory:")
    tbl_schema = DBInterface.execute(con, "DESCRIBE TABLE '$path'") |> DataFrames.DataFrame
    keys = tbl_schema[:, :column_name]
    uniques = unique(keys)
    if length(keys) > length(uniques)
        dupes = keep_duplicated_entries!(keys, uniques)
        @error """Some duplicate column names detected in the header of $path: $dupes.
        Duplicate column names may cause errors, as only the first is used.
        """
    end
end
```